plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1391,6 +1442,92 @@ private List<ColumnMetadata> getColumnMetadatas(Schema schema)
         return columns.build();
     }
 
+    @Override
+    public ConnectorAnalyzeMetadata getStatisticsCollectionMetadata(ConnectorSession session, ConnectorTableHandle tableHandle, Map<String, Object> analyzeProperties)
+    {
+        IcebergTableHandle handle = (IcebergTableHandle) tableHandle;
+        checkArgument(handle.getTableType() == DATA, "Cannot analyze non-DATA table: %s", handle.getTableType());
```

Comment:
Should `getTableHandleForExecute` have a similar check?

Reply:
Also, do we verify that the table's snapshot is the recent one? Not time traveling

Reply:
Added both